artifactory/commands/npm/publish.go

```diff
@@ -202,22 +220,32 @@ func (npc *NpmPublishCommand) doDeploy(target string, artDetails *config.ServerD
 	}
 	up := services.UploadParams{}
 	up.ArtifactoryCommonParams = &specutils.ArtifactoryCommonParams{Pattern: npc.packedFilePath, Target: target}
-	if npc.collectBuildInfo {
-		utils.SaveBuildGeneralDetails(npc.buildConfiguration.BuildName, npc.buildConfiguration.BuildNumber, npc.buildConfiguration.Project)
-		up.BuildProps, err = utils.CreateBuildProperties(npc.buildConfiguration.BuildName, npc.buildConfiguration.BuildNumber, npc.buildConfiguration.Project)
-		if err != nil {
-			return err
-		}
-	}
 	var totalFailed int
-	if npc.collectBuildInfo {
+	if npc.collectBuildInfo || npc.detailedSummary {
+		if npc.collectBuildInfo {
+			utils.SaveBuildGeneralDetails(npc.buildConfiguration.BuildName, npc.buildConfiguration.BuildNumber, npc.buildConfiguration.Project)
+			up.BuildProps, err = utils.CreateBuildProperties(npc.buildConfiguration.BuildName, npc.buildConfiguration.BuildNumber, npc.buildConfiguration.Project)
+			if err != nil {
+				return err
+			}
+		}
 		summary, err := servicesManager.UploadFilesWithSummary(up)
 		if err != nil {
 			return err
 		}
-		summary.TransferDetailsReader.Close()
-		npc.artifactsDetailsReader = summary.ArtifactsDetailsReader
 		totalFailed = summary.TotalFailed
+		if npc.collectBuildInfo {
+			npc.artifactsDetailsReader = summary.ArtifactsDetailsReader
+		} else {
+			defer summary.ArtifactsDetailsReader.Close()
+		}
+		if npc.detailedSummary {
+			npc.result.SetReader(summary.TransferDetailsReader)
+			npc.result.SetFailCount(totalFailed)
+			npc.result.SetSuccessCount(summary.TotalSucceeded)
+		} else {
+			defer summary.TransferDetailsReader.Close()
```

Comment:
Can't it be without defer?
```suggestion
			summary.TransferDetailsReader.Close()
```